st2common/st2common/models/api/rule.py

```diff
@@ -235,6 +241,17 @@ def to_model(cls, rule):
         trigger_type_ref = trigger.get('type', None)
         parameters = trigger.get('parameters', {})
 
+        context = {}
+        context.update({
+            kv_constants.DATASTORE_PARENT_SCOPE: {
+                kv_constants.SYSTEM_SCOPE: kv_service.KeyValueLookup(
+                    scope=kv_constants.FULL_SYSTEM_SCOPE)
+            }
+        })
+        parameters = jinja_utils.render_values(mapping=parameters, context=context,
```

Comment:
By rending in the RuleAPI, will this render and expose secrets when getting the rule from the CLI?

Reply:
I prefer the rendering is done somewhere else. When getting the rule via API/CLI, users should see the expressions. 1) It won't confuse user with what's actually is defined for the rule 2) Secrets if any are not displayed.

Reply:
@m4dcoder We already discussed this in the issue ^ and came to the decision that we want to render in the rule registration. We simply cannot render the parameters later because the rule needs to be registered with type and actual parameters. Rule matching happens with this info. 

I see the problems with secret however. Unfortunately, our rule model doesn't have a `secret` metadata flag for parameters. So we'd expose secrets. Good catch. To "fix" this, we'd have to update rule model. @dzimine @nmaludy This is getting a little hairy.